### PR TITLE
crash fix during server failover

### PR DESCRIPTION
**Pull request by andreygolev**
This is not the best implementation, but fixes crash when worker agent after job execution tries to send ExecutionDone to a server in a cluster, but server is already dead.

Don't know how to do it better than net.Dial() for now.
As an addition it would be imho better to add retries with different members

### CODE REVIEW

File: dkron/invoke.go [hunk 1]
```diff
@@ -61,7 +62,7 @@ func (a *Agent) invokeJob(job *Job, execution *Execution) error {
 	execution.Success = success
 	execution.Output = output.Bytes()
 
-	rpcServer, err := a.selectServerByKey(execution.Key())
```
Comment (victorcoder) [inline dkron/invoke.go]: can `selectServerByKey` be removed?

Comment (andreygolev) [reply dkron/invoke.go]: according to the code, there are no occurrences.

**Comment by victorcoder** [comment]
@andreygolev did you experienced crashes with the old method? I don't get the point of doing a net.Dial check, the `LocalServers` method already returns the servers available that are live. The timming is the same as in this proposal.

**Comment by andreygolev** [comment]
Yes, I've experienced crashes with the old method.
It was clearly visible in debug logs that worker agent was trying to send RPC call to an IP of an already dead server.
LocalServer can return a node that in reality is already dead, but fact still not yet confirmed by cluster, because failover takes some time.

net.Dial just checks if you can connect to the IP:port. If not, then try next one.

**Comment by andreygolev** [comment]
Better approach would be to add retries imho

File: dkron/invoke.go [hunk 2]
```diff
@@ -72,18 +72,21 @@ func (a *Agent) invokeJob(job *Job, execution *Execution) error {
...
-		ch.Add(p.RPCAddr.String())
+		peers = append(peers, p.RPCAddr.String())
 	}
 
-	peerAddress := ch.Get(key)
-	if peerAddress == "" {
-		return "", ErrNoSuitableServer
+	for _, peer := range peers {
+		log.Debugf("Checking peer: %v", peer)
+		conn, err := net.DialTimeout("tcp", peer, 5*time.Second)
```
Comment (victorcoder) [inline dkron/invoke.go]: 5 seconds for timeout seems too much

Comment (andreygolev) [reply dkron/invoke.go]: Safe choice :)
What about 1 second?

Comment (victorcoder) [reply dkron/invoke.go]: It should work well

Comment (andreygolev) [reply dkron/invoke.go]: here we go

**Pull request merged**